modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/GrokProcessorGetAction.java

```diff
@@ -0,0 +1,167 @@
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.action.RestBuilderListener;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.ingest.common.IngestCommonPlugin.GROK_PATTERNS;
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+import static org.elasticsearch.rest.RestStatus.OK;
+
+public class GrokProcessorGetAction extends Action<GrokProcessorGetAction.Request,
+    GrokProcessorGetAction.Response, GrokProcessorGetAction.RequestBuilder> {
+
+    public static final GrokProcessorGetAction INSTANCE = new GrokProcessorGetAction();
+    public static final String NAME = "cluster:admin/ingest/processor/grok/get";
+
+    private GrokProcessorGetAction() {
+        super(NAME);
+    }
+
+    @Override
+    public RequestBuilder newRequestBuilder(ElasticsearchClient client) {
+        return new RequestBuilder(client);
+    }
+
+    @Override
+    public Response newResponse() {
+        return new Response();
+    }
+
+    public static class Request extends ActionRequest {
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+    }
+
+    public static class RequestBuilder extends ActionRequestBuilder<Request, Response, RequestBuilder> {
+        public RequestBuilder(ElasticsearchClient client) {
+            super(client, GrokProcessorGetAction.INSTANCE, new Request());
+        }
+    }
+
+    public static class Response extends AcknowledgedResponse implements ToXContentObject {
+        private Map<String, String> grokPatterns;
+
+        Response() {
+
+        }
+
+        public Response(Map<String, String> grokPatterns) {
+            this.grokPatterns = grokPatterns;
+        }
+
+        public Map<String, String> getGrokPatterns() {
+            return grokPatterns;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("patterns");
+            builder.map(grokPatterns);
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            grokPatterns = (Map<String, String>) in.readGenericValue();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeMapWithConsistentOrder(grokPatterns);
```

Comment:
Not sure why the order matters. Since we know this is a string -> string map, let's write it for that:
`out.writeMap(grokPatters, StreamOutput::writeString, StreamOutput::writeString)`